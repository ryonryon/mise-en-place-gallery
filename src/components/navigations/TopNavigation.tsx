import styled from "@emotion/styled";
import * as React from "react";

import { white } from "../../constants/colors";

interface Props {
  logo: string;
  navHeight: string;
  items: string[];
}

const TopNavigation = ({ logo, navHeight, items }: Props) => {
  const StyledNav = styled.nav`
    width: 100%;
    height: ${navHeight};
    background-color: #323439;
    color: #e5e5e5;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: space-between;

    .nav {
      &:hover {
        color: ${white};
        box-shadow: 1px 1px 4px black;
      }
    }

    .logo {
      padding: 14px 24px 14px 24px;
      font-size: 24px;
    }

    .options {
      display: flex;
    }

    .option {
      padding: 19px;
    }

    & a {
      text-decoration: none;
      color: ${white};
    }
  `;

  return (
    <StyledNav>
      <a href="">
        <div className="nav logo">{logo}</div>
      </a>
      <div className="options">
        {items.map((item, index) => (
          <a key={index} href={item.toLowerCase()}>
            <div className="nav option">{item}</div>
          </a>
        ))}
      </div>
    </StyledNav>
  );
};

TopNavigation.defaultProps = {
  navHeight: "56px"
};

export default TopNavigation;
