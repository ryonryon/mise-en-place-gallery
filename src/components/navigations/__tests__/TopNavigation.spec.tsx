import * as React from "react";
import * as renderer from "react-test-renderer";

import TopNavigation from "../TopNavigation";

describe("TopNavigation test", () => {
  it("tests TopNavigation render correctly", () => {
    const topNavigation = renderer.create(<TopNavigation logo="test" items={["test1", "test2", "test3", "test4"]} />);

    expect(topNavigation).toMatchSnapshot();
  });
});
