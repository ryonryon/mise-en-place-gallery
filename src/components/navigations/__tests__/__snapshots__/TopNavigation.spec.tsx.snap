// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavigation test tests TopNavigation render correctly 1`] = `
<nav
  className="css-y95ww1"
>
  <a
    href=""
  >
    <div
      className="nav logo"
    >
      test
    </div>
  </a>
  <div
    className="options"
  >
    <a
      href="test1"
    >
      <div
        className="nav option"
      >
        test1
      </div>
    </a>
    <a
      href="test2"
    >
      <div
        className="nav option"
      >
        test2
      </div>
    </a>
    <a
      href="test3"
    >
      <div
        className="nav option"
      >
        test3
      </div>
    </a>
    <a
      href="test4"
    >
      <div
        className="nav option"
      >
        test4
      </div>
    </a>
  </div>
</nav>
`;
